modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3725,6 +3725,8 @@ def to_pandas(self):
         df = self._partition_mgr_cls.to_pandas(self._partitions)
         if df.empty:
             df = pandas.DataFrame(columns=self.columns, index=self.index)
+            if len(df.columns) and self.has_materialized_dtypes:
+                df = df.astype(self.dtypes)
```

Comment:
I wonder why to fix `copy` on empty frames we make changes to `to_pandas()`... and why cannot we do an unconditional explicit `pandas.DataFrame(..., dtypes=self.dtypes)` instead of casting with `.astype()`

Reply:
> I wonder why to fix copy on empty frames we make changes to to_pandas()

In order not to complicate the code with edge cases, almost all methods called on empty objects are defaulted to pandas.

> and why cannot we do an unconditional explicit pandas.DataFrame(..., dtypes=self.dtypes) instead of casting with .astype()

This is a constructor constraint - https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html?highlight=dataframe#pandas.DataFrame
```python
dtype : dtype, default None
    Data type to force. Only a single dtype is allowed. If None, infer.
```